src/errors.ml

```diff
@@ -75,14 +75,35 @@ let print ppf loc =
           Format.fprintf pp "%s\n%*s\n" line
             stop_c
             (String.make len '^')
-        else if num_lines <= 10 then
-          let lines = file_lines path ~start:start.pos_lnum ~stop:stop.pos_lnum in
-          let last_lnum = Option.map ~f:fst (List.last lines) in
-          let padding_width = Option.value_exn
-                                (Option.map ~f:String.length last_lnum) in
-          List.iter ~f:(fun (lnum, l) ->
-            Format.fprintf pp "%*s: %s\n" padding_width lnum l)
-          lines
+        else
+          let get_padding lines =
+            let last_lnum = Option.map ~f:fst (List.last lines) in
+            Option.value_exn (Option.map ~f:String.length last_lnum)
+          in
+          let print_lines ?(trailing_line=false) lines padding_width =
+            List.iter ~f:(fun (lnum, l) ->
+              Format.fprintf pp "%*s | %s\n" padding_width lnum l) lines;
+            if trailing_line then
+              (* We add 2 to the width of max line to account for
+                 the extra space and the `|` character at the end
+                 of a line number *)
+              let line = String.make (padding_width + 2) '.' in
+              Format.fprintf pp "\n%s\n\n" line
+          in
+          if num_lines <= 10 then
```

Comment:
What do you think about extracting this constant out of this function ? Like `let maximum_lines_to_print_in_full` or something. Same goes for the `3` below, you can pull out a `let context_lines = 4`

Reply:
good point. I'll address this